### PR TITLE
Fix golang purls

### PR DESCRIPTION
**Pull request by sbs2001**
Fix https://github.com/kubernetes-sigs/bom/issues/169

### CODE REVIEW

**Comment by cpanato** [comment]
/assign @puerco 

we need to review this, I personally would not like to have syft as dependency here

File: go.mod [hunk 1]
```diff
@@ -3,40 +3,71 @@ module sigs.k8s.io/bom
 go 1.18
 
 require (
+	github.com/anchore/syft v0.58.0
```
Comment (puerco) [inline go.mod:6]: OK, not like this. This is pulling the whole syft project as a dependency.

**Review by puerco — CHANGES_REQUESTED** [review]
Thanks for this @sbs2001 
We do need to generate CPEs for go modules. But this PR is pulling all of syft as a dependency for achieve. We need to implement our own CPE logic or find a specialized library that will do that for us.

**Comment by sbs2001** [comment]
@puerco  this PR now only fixes https://github.com/kubernetes-sigs/bom/issues/169

**Comment by puerco** [comment]
Thanks @sbs2001 Sorry I was (am) traveling and I missed the last updates to this PR. It LGTM in general but it seems the linter is not happy. Also, could you add a simple unit test to ensure we don't miss this one again?

Thanks!

**Comment by sbs2001** [comment]
@puerco  I'll make changes later this week or next week (I have exams rn)

**Comment by puerco** [comment]
No rush, if i can I'll try to push the linter fixes into your branch to help out!

**Comment by sbs2001** [comment]
@puerco  linter is fixed and test is added

**Review by puerco — APPROVED** [review]
Thanks @sbs2001 !
/lgtm

**Pull request merged**